### PR TITLE
JAMES-2665 Integration tests for DeletedMessageVault - Search

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 1]
```diff
@@ -483,7 +483,9 @@ private void restoreAllMessagesOfHomer() {
 
     private void restoreMessagesFor(String user) {
         String taskId = webAdminApi.with()
-            .post("/deletedMessages/user/" + user + "?action=restore")
+            .queryParam("action", "restore")
+            .body("{\"criterions\":[]}")
+            .post("/deletedMessages/user/" + user).prettyPeek()
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: `prettyPeek()` ?

Comment (chibenwa) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: I'm NEVER letting debug statement make it to the review... :trollface:

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

```
[dfb6562fde8b305996f22e14847ad7a99c276a3c] [ERROR]   EventDeadLettersContract$RemoveContract.removeShouldKeepConsistencyWhenConcurrentRemove:276 
[dfb6562fde8b305996f22e14847ad7a99c276a3c] Expecting empty but was:<[EventId{id=c3737794-a110-4d40-a9af-1157fff04a0e}]>
```

**Comment by Arsnael** [comment]
Lots of issues with guice injection, please check

**Comment by chibenwa** [comment]
> Lots of issues with guice injection, please check

When you leave such comments, copy and pasting a part of the error message would be much appreciated.

Build result lost...

File: server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/query/QueryTranslator.java [hunk 2]
```diff
@@ -164,6 +165,7 @@ static MailAddress parseMailAddress(String mailAddressString) throws QueryTransl
 
     private final ImmutableTable<FieldName, Operator, Function<String, Criterion>> criterionRegistry;
 
+    @Inject
     public QueryTranslator(MailboxId.Factory mailboxIdFactory) {
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/query/QueryTranslator.java]: should not be public

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/query/QueryTranslator.java]: Well, this constructor is accessed bu the route tests located in another package

Moving it to the same package creates much conflicts (DTO package-private constructors)

The only solution might be fusing the two packages. What I don't like.

Comment (mbaechler) [reply server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/query/QueryTranslator.java]: use @VisibleForTesting + public then

**Comment by chibenwa** [comment]
Pfff

```
RabbitMQIMAPInconsistenciesTest » ConditionTimeout Condition with lambda expre...
```

Will get a look tomorrow

test this please

**Comment by mbaechler** [comment]
I'm already investigating rabbit thread leak, it's probably the responsible for our current build instability

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 3]
```diff
@@ -208,6 +212,72 @@ public void vaultEndpointShouldRestoreImapDeletedMailbox() throws Exception {
...
+
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 1);
+
+        String messageId = listMessageIdsForAccount(homerAccessToken).get(0);
+        given()
+            .header("Authorization", homerAccessToken.serialize())
+            .body("[[\"getMessages\", {\"ids\": [\"" + messageId + "\"]}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+            .then()
```
Comment (Arsnael) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: Ident ?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java [hunk 4]
```diff
@@ -416,6 +490,21 @@ private void restoreAllMessagesOfHomer() {
...
         String taskId = webAdminApi.with()
+            .queryParam("action", "restore")
+            .body(MATCH_ALL_QUERY)
+            .post("/deletedMessages/user/" + user)
+            .jsonPath()
+            .get("taskId");
+
+        webAdminApi.given()
+            .get("/tasks/" + taskId + "/await")
+            .then()
```
Comment (Arsnael) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java]: Idem

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**